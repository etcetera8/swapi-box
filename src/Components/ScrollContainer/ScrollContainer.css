.ScrollContainer {
  background: gray;
  width: 100%;
  height: 100px;
}