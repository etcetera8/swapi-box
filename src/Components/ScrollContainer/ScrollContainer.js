import React from 'react';
import PropTypes from 'prop-types';
import './ScrollContainer.css';

const ScrollContainer = ( {movieData} ) => {
  const {scrollText, episode, title, releaseDate} = movieData;
  
  return (
    <section className='ScrollContainer'>
      <div className="crawl">
        <h1 className="swapi-title">SWAPI-BOX </h1>
        <br />
        <p>{scrollText}</p>
        <h3>{releaseDate}</h3>
        <h2>{title}</h2>
        <h3>Episode: {episode}</h3>
      </div>
    </section>
  )
}

ScrollContainer.propTypes = {
  movieData: PropTypes.shape({
    scrollText: PropTypes.string,
    episode: PropTypes.string,
    title: PropTypes.string,
    releaseDate: PropTypes.string
  })
};

export default ScrollContainer;