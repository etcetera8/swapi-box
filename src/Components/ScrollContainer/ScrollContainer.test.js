import React from 'react';
import ReactDOM from 'react-dom';
import ScrollContainer from './ScrollContainer';
import { shallow } from 'enzyme';

describe('ScrollContainer', () => {
  let wrapper;
  const mockFunction = jest.fn();
  
  beforeEach( () => {
    wrapper = 
      shallow(<ScrollContainer 
                movieData={
                  {scrollText: "once upon a time",
                  episode:"v", 
                  title:"A new Hope", 
                  releaseDate:"05-1985"
                  }}
              />)
  })

  it('should match the snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  })
})