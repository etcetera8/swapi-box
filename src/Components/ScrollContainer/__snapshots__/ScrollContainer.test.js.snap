// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollContainer should match the snapshot test 1`] = `
<section
  className="ScrollContainer"
>
  <div
    className="crawl"
  >
    <h1
      className="swapi-title"
    >
      SWAPI-BOX 
    </h1>
    <br />
    <p>
      once upon a time
    </p>
    <h3>
      05-1985
    </h3>
    <h2>
      A new Hope
    </h2>
    <h3>
      Episode 
      v
    </h3>
  </div>
</section>
`;
